### PR TITLE
Refactor node UI

### PR DESCRIPTION
**Pull request by jdent02**

### CODE REVIEW

**Comment by jdent02** [comment]
This one is ready for review I think

**Comment by jdent02** [comment]
I don't think you're looking at the latest revision.....

**Comment by dictoon** [comment]
Hm, I just checked and yes I'm looking at the latest revision... :) Which comments aren't relevant in your opinion?

**Comment by jdent02** [comment]
These ones:
image

**Comment by jdent02** [comment]
This one:

image

The file actually looks like this:

```python
for y in socket_input_names:
                input = self.inputs.new(y['socket_name'], y['socket_label'])
                if (y['connectable'] is True and y['hide_ui'] is not True) or y['connectable'] is False:
                    input.hide = True
```

**Comment by jdent02** [comment]
As far as the socket color question, I'm hamstrung by the way Blender works.  Each socket class needs a
```python
def draw_color(self, context, node)
```
function.  It can only take those arguments.  I could factorize it probably but it would be much more convoluted.  The way I have takes up more screen space but it's conceptually more straightforward.

**Comment by jdent02** [comment]
I've addressed your comments on the 'x' and 'y' variables not being well named.

**Comment by dictoon** [comment]
Can you please merge related commits together?

**Comment by jdent02** [comment]
I'll work on it on a bit.  I have an addition or two to make.

**Comment by jdent02** [comment]
Oh, and to answer your backwards compatibility question: from all of my tests backwards compatibility is maintained.  At file load a new util function called 'update_project' runs.  It copies any modified socket values into the identical value in the node.  In the future this util function could also make other updates if we change additional things.

File: properties/nodes/oslnode.py [hunk 1]
```diff
@@ -164,7 +164,7 @@ def draw_buttons(self, context, layout):
         socket_number = 0
         param_section = ""
         for x in input_params:
```
Comment (dictoon) [inline properties/nodes/oslnode.py]: Another place where `x` is used for iteration.

**Pull request merged**